### PR TITLE
COMP: Add sources for ITK modules only if they are enabled.

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
Cc: @thewtex

File: include/CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,6 @@ macro( add_subdirectory_if_module_enabled _dir )
   endif()
 endmacro()
 
-add_subdirectory( Core )
-add_subdirectory( Filtering )
-add_subdirectory( Numerics )
+add_subdirectory_if_module_enabled( Core )
+add_subdirectory_if_module_enabled( Filtering )
+add_subdirectory_if_module_enabled( Numerics )
```
Comment (thewtex) [inline include/CMakeLists.txt]: These should still be `add_subdirectory` -- `Core`, `Filtering`, and `Numerics` are Groups as opposed to Modules.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,22 @@ set( ${PROJECT_NAME}_SRCS
...
+# Utility macro: add sources if the corresponding module is enabled
+# when ITK was built.
+macro( add_sources_if_module_enabled _dir _sources )
+  list( FIND ITK_MODULES_ENABLED ITK${_dir} _have )
+  if( NOT ${_have} EQUAL "-1" )
+    list( APPEND ${PROJECT_NAME}_SRCS ${_sources} )
+  endif()
+endmacro()
+
+add_sources_if_module_enabled( Numerics
```
Comment (thewtex) [inline src/CMakeLists.txt]: :+1:

**Comment by jcfr** [comment]
Thanks for the review.

**Pull request merged**